### PR TITLE
use peppy

### PR DESCRIPTION
**Pull request by stolarczyk**
@vreuter, @nsheff, @johanneskoester

This is a reiteration of #8, but with a re-written version of `peppy` that supports the config format revamp discussed in https://github.com/pepkit/peppy/issues/334

### CODE REVIEW

File: rules/common.smk [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 from snakemake.utils import validate
 from snakemake.utils import min_version
 
 min_version("5.7.1")
 
 report: "../report/workflow.rst"
 
+p = Project(cfg="prjcfg.yaml", sample_table_index="sample",
+            subsample_table_index=["sample", "unit"])
+
```
Comment (stolarczyk) [inline rules/common.smk]: `sample_table_index` and `subsample_table_index` arguments could be skipped if snakemake followed `peppy` defaults: 

- `"sample_name"` for sample_table index colnames
- `["sample_name", "subsample_name"]` for subsample_table index colnames

**Comment by nsheff** [comment]
@stolarczyk can you update this to the now-almost-final peppy updates when you get a chance?

**Comment by nsheff** [comment]
FYI we released peppy so it's now live with all the updates we discussed.

**Comment by johanneskoester** [comment]
Hi Nathan,
yes, I'll have a look. Still not a fan of the snake_config tag, but maybe it can be used in a different way. I just cannot promise that I will be able to evaluate it immediately.

Best,
Johanne

**Comment by nsheff** [comment]
> Still not a fan of the snake_config tag

that's not build in to peppy -- it's just used here. you could instead just put all options in `config.yaml` directly into the PEP... is that what you're wanting?

**Comment by johanneskoester** [comment]
Thanks for the clarification! I will probably keep the PEP and the pipeline config separate then. My idea is that the PEP should just describe the samples, while the pipeline describes the analysis. This way, one can easily plug the very same PEP into a completely different pipeline (other WMS, other analysis, ...).

**Pull request closed without merging**